src/main/java/org/commcare/suite/model/Detail.java

```diff
@@ -443,4 +459,74 @@ public boolean evaluateFocusFunction(EvaluationContext ec) {
     public XPathExpression getFocusFunction() {
         return focusFunction;
     }
+
+    private boolean templatePathValid(String templatePathProvided) {
+        if (PRINT_TEMPLATE_PROVIDED_VIA_GLOBAL_SETTING.equals(templatePathProvided)) {
+            return true;
+        } else if (templatePathProvided != null) {
+            try {
+                this.derivedPrintTemplatePath =
+                        ReferenceManager.instance().DeriveReference(templatePathProvided).getLocalURI();
+                return true;
+            } catch (InvalidReferenceException e) {
+                System.out.println("Invalid print template path provided for detail with id " + this.id);
+            }
+        }
+        return false;
+    }
+
+    private boolean detailFieldsHaveIds() {
+        for (DetailField field : this.fields) {
+            if (field.getPrintIdentifierRobust() == null) {
+                return false;
```

Comment:
Wouldn't it be fine for some fields to not have an identifier if it wasn't actually needed by the template? I would have thought only field detail items included inside of the template would  *need* an ID.

I guess I was assuming that the presence of the Template was the "cue" for printing, and that if anything was missing to make it work we'd fail with an error (IE "Detail is missing required template field 'ID') rather than silently.